lang/src/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -77,6 +80,8 @@ typealias ProcessingStep = (PartiqlAst.Statement, StepContext) -> PartiqlAst.Sta
  * per thread.
  */
 interface CompilerPipeline {
+    val scalarTypeSystem: ScalarTypeSystem
```

Comment:
We consider a scalar type system as one configuration option of the compiler pipeline.